### PR TITLE
[fetcheus] update angle-vector and add angle-vector-sequence

### PR DESCRIPTION
**Pull request by knorth55**
update `fetch` `:angle-vector` to follow `baxter` `:angle-vector` with `moveit`

### CODE REVIEW

**Comment by k-okada** [comment]
could you explain whey we we need to update these method?

**Comment by knorth55** [comment]
I set default values for `:angle-vector` method.
And fetch-users also want to use `:angle-vector-sequence` with `MoveIt!`

**Comment by k-okada** [comment]
in that case, please consider putting these functions in robot-interface.l and fetch-interface.l can use these features with minimum changes.

**Comment by knorth55** [comment]
this method is actually modified for `fetch`.
these cannot be applied to other robots.

**Comment by k-okada** [comment]
it seems we have several improvement in this PR

1. eable to use angle-vector-sequence -> ok
2. use ctype?
3. enable to use angle-vector-raw if time is nil -> I think we will not support a way to call :angle-vector-raw implicitly for fetch robot.

**Comment by knorth55** [comment]
> 2. use ctype?

these changes are needed to adapt to default `:angle-vector` method.
default `:angle-vector` pass `ctype` in `&optional`, but `:angle-vector-motion-plan` pass `ctype` in `&key`

> 3. enable to use angle-vector-raw if time is nil -> I think we will not support a way to call :angle-vector-raw implicitly for fetch robot.

if `tm` is `nil`, use `angle-vector-motion-plan` with `tm` `3000`

**Comment by knorth55** [comment]
Should I split this commit into two?

**Comment by knorth55** [comment]
Updated

File: jsk_fetch_robot/fetcheus/fetch-interface.l [hunk 1]
```diff
@@ -25,24 +25,42 @@
...
-   (let ((use-torso t))
-     (if (and (member :use-torso args) (null (cadr (member :use-torso args)))) (setq use-torso nil))
-     (send self :angle-vector-motion-plan av :move-arm :rarm :total-time tm :start-offset-time 0.01 :use-torso use-torso :clear-velocities t)
-   ))
-  ;;
+   (send* self :angle-vector-motion-plan av :ctype ctype :move-arm :rarm :total-time tm
+               :start-offset-time start-offset-time :clear-velocities clear-velocities
+               :use-torso use-torso args))
+  (:angle-vector-sequence
+   (avs &optional (tm 3000) (ctype controller-type) (start-time 0) &rest args
```
Comment (k-okada) [inline jsk_fetch_robot/fetcheus/fetch-interface.l]: :angle-vector-sequence takes `tms`, list of time
https://github.com/jsk-ros-pkg/jsk_pr2eus/blob/0.3.10/pr2eus/robot-interface.l#L400

File: jsk_fetch_robot/fetcheus/fetch-interface.l [hunk 2]
```diff
@@ -25,24 +25,42 @@
...
+        &key (use-torso t) (start-offset-time 0.01) (clear-velocities t) &allow-other-keys)
+   "Send joind angle to robot with self-collision motion planning, this method returns immediately, so use :wait-interpolation to block until the motion stops.
+- avs : sequence of joint angle vector [rad]
+- tm : time to goal in [msec]
```
Comment (k-okada) [inline jsk_fetch_robot/fetcheus/fetch-interface.l]: `tms: list of time to goal from previous angle-vector point in [msec]`?

File: jsk_fetch_robot/fetcheus/fetch-interface.l [hunk 3]
```diff
@@ -25,24 +25,42 @@
...
+- use-torso : set t to use torso
+"
+   (setq ctype (or ctype controller-type))  ;; use default if ctype is nil
+   ;; for simulation mode
+   (when (send self :simulation-modep)
+     (return-from :angle-vector-sequence
+                  (send* self :angle-vector-sequence-raw avs tm ctype start-time args)))
+   (when (not (numberp tm))
+     (ros::warn ":angle-vector tm is not a number, use :angle-vector av tm args"))
+   (send* self :angle-vector-motion-plan avs :ctype ctype :move-arm :rarm :total-time tm
```
Comment (k-okada) [inline jsk_fetch_robot/fetcheus/fetch-interface.l]: `apply #'+ tms` ?

File: jsk_fetch_robot/fetcheus/fetch-interface.l [hunk 4]
```diff
@@ -25,24 +25,42 @@
...
+- avs : sequence of joint angle vector [rad]
+- tm : time to goal in [msec]
+- use-torso : set t to use torso
+"
+   (setq ctype (or ctype controller-type))  ;; use default if ctype is nil
+   ;; for simulation mode
+   (when (send self :simulation-modep)
+     (return-from :angle-vector-sequence
+                  (send* self :angle-vector-sequence-raw avs tm ctype start-time args)))
+   (when (not (numberp tm))
```
Comment (k-okada) [inline jsk_fetch_robot/fetcheus/fetch-interface.l]: `every #'numberp tms` ?

File: jsk_fetch_robot/fetcheus/fetch-interface.l (hunk 1 above)
Comment (knorth55) [inline jsk_fetch_robot/fetcheus/fetch-interface.l]: Should I modify `pr2eus_moveit` `:angle-vector-motion-plan` to support `tms` as list of time?

**Comment by knorth55** [comment]
Updated

**Pull request closed without merging**